pkg/reconciler/shared/tektonconfig/pipeline/pipeline.go

```diff
@@ -20,35 +20,51 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"log"
 	"reflect"
 
 	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
 
 	op "github.com/tektoncd/operator/pkg/client/clientset/versioned/typed/operator/v1alpha1"
-	operatorv1alpha1 "github.com/tektoncd/operator/pkg/client/clientset/versioned/typed/operator/v1alpha1"
 	"github.com/tektoncd/operator/pkg/reconciler/common"
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"knative.dev/pkg/test/logging"
 )
 
-func CreatePipelineCR(ctx context.Context, instance v1alpha1.TektonComponent, client operatorv1alpha1.OperatorV1alpha1Interface) error {
-	configInstance := instance.(*v1alpha1.TektonConfig)
-	if _, err := ensureTektonPipelineExists(ctx, client.TektonPipelines(), configInstance); err != nil {
-		return errors.New(err.Error())
+func EnsureTektonPipelineExists(ctx context.Context, clients op.TektonPipelineInterface, config *v1alpha1.TektonConfig) (*v1alpha1.TektonPipeline, error) {
+	tpCR, err := GetPipeline(ctx, clients, v1alpha1.PipelineResourceName)
+
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			tpCR = &v1alpha1.TektonPipeline{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: v1alpha1.PipelineResourceName,
+				},
+				Spec: v1alpha1.TektonPipelineSpec{
+					CommonSpec: v1alpha1.CommonSpec{
+						TargetNamespace: config.Spec.TargetNamespace,
+					},
+					Pipeline: config.Spec.Pipeline,
+					Config:   config.Spec.Config,
+				},
+			}
+			_, err = clients.Create(ctx, tpCR, metav1.CreateOptions{})
```

Comment:
just an observation, here we have a separate function for 1 line get operation but not separate function for the multiline create operation. 

If we could extract the create logic (initialize struct and call create Pipeline api) into a function, this code block will have better flow. 🧑‍💻